docs/content/new-tutorial/building-an-asset-graph.mdx

```diff
@@ -0,0 +1,221 @@
+
+    return df
+```
+
+For more information about loggers, such as the different types of logging, you can read the [logging documentation](/concepts/logging/loggers).
+
+---
+
+## Creating and exporting the visualization asset
+
+The last asset you'll add is the visualization asset. This asset will take the dataframe asset and create a visualization of the data. You can copy most of the last chunk of code from the original pipeline. Once again, we'll also add the relevant imports.
+
+```python
+import matplotlib.pyplot as plt
+import base64
+from io import BytesIO
+from wordcloud import STOPWORDS, WordCloud
+from dagster import asset
+
+@asset
+def hackernews_topstories_word_cloud(context, hackernews_topstories):
+    stopwords = set(STOPWORDS)
+    stopwords.update(["Ask", "Show", "HN"])
+    titles_text = " ".join([str(item) for item in hackernews_topstories["title"]])
+    titles_cloud = WordCloud(stopwords=stopwords, background_color="white").generate(titles_text)
+
+    # Generate the word cloud image
+    plt.figure(figsize=(8, 8), facecolor=None)
+    plt.imshow(titles_cloud, interpolation="bilinear")
+    plt.axis("off")
+    plt.tight_layout(pad=0)
+
+    # Save the image to a buffer
+    buffer = BytesIO()
+    plt.savefig(buffer, format="png")
+    image_data = base64.b64encode(buffer.getvalue())
+
+    return image_data
+```
+
+Right now, materializing the asset will save the file to local storage. In a real-world scenario, you would save the file to a cloud storage bucket. In a later section, you'll learn how to tell Dagster where to store the data by using IO managers.
+
+### Metadata and previews
+
+As mentioned, the asset is materialized in storage. You can add metadata to help users explore the asset and understand it. For example, you can add a preview of the asset to the metadata.
+
+By modifying your code to match the snippet below, you will add metadata to your asset.
+
+```python
+from dagster import MetadataValue, Output, asset
+
+@asset
+def hackernews_topstories_word_cloud(context, hackernews_topstories):
+    stopwords = set(STOPWORDS)
+    stopwords.update(["Ask", "Show", "HN"])
+    titles_text = " ".join([str(item) for item in hackernews_topstories["title"]])
+    titles_cloud = WordCloud(stopwords=stopwords, background_color="white").generate(titles_text)
+
+    # Generate the word cloud image
+    plt.figure(figsize=(8, 8), facecolor=None)
+    plt.imshow(titles_cloud, interpolation="bilinear")
+    plt.axis("off")
+    plt.tight_layout(pad=0)
+
+    # Save the image to a buffer and embed the image into Markdown content for quick view
+    buffer = BytesIO()
+    plt.savefig(buffer, format="png")
+    image_data = base64.b64encode(buffer.getvalue())
+
+    md_content = f"![img](data:image/png;base64,{image_data.decode()})"
+
+    # Attach the Markdown content as metadata to the asset
+    return Output(value=image_data, metadata={
+        "plot": MetadataValue.md(md_content)
+    })
+```
+
```

Comment:
🤔 One thing I think this section is missing is telling users where they can find metadata, especially in Dagit. This bit tells users what the metadata does ("adds a preview") but not where they can check it out. I think it's worthwhile to add this info.